Broke up create_text_sprite into 2 functions [WIP][HELP]

gran4 opened the pull request:
Fixed https://github.com/pythonarcade/arcade/issues/1410

Help me fix multiline issue(everything but multiline works).

Commit:
Broke Up create_text_sprite

https://github.com/pythonarcade/arcade/issues/1410

Commit:
Changed test

einarf commented:
There was some challenges related to implementing these changes I can't remember at the moment. Will have to look closer at this one.

Also: Seems your unit tests are failing. Did you run them locally before pushing?

gran4 commented:
I had to add create_text_texture to __all__.

Commit:
Had to add create_text_texture to __all__

Commit:
Fixed width and height bug.

normal width and height is bugged on the label, have to use content_width and content_height

Commit:
Small change

Commit:
warning was breaking

Commit:
Warning should be fixed

gran4 committed:
Update text.py

Add/Change comments on functions

gran4 commented:
I think you guys can review. @einarf

eschan145 commented on arcade/text.py:
```diff
@@ -580,17 +579,13 @@ def create_text_sprite(
 
-    It then creates a sprite referencing the newly created texture, and positions it
-    accordingly, and that is final result that is returned from the function.
+    Internally this creates a pyglet.text.Label object and an empty texture. It then uses either the
```
Use this instead:
```
:py:class:`~pyglet.text.label`
```

This will yield RST formatted text. Also, try to wrap this line to eighty characters to make it more human-readable.

Commit:
Fix @eschan145 's review

Commit:
Fix whitespaces in text

gran4 commented:
@einarf I think you can merge this. Is there any problem with it?

einarf commented:
I will look at it when back from hospital unless someone else beats me to it. It might linger here for a while,

einarf commented:
I'm wondering if we should try delay this feature after 3.0.0 when other related systems are in better shape. When I tested this last time there was some text getting cutt off as well in certain enviroments. Either afer 3.0.0 or closer to release.

gran4 commented:
@einarf It doesn't really deal with any of the lower level stuff. All it is suppose to do is allow you to get the texture in certain use cases like issue #1410. If the text was cut off, it is either a problem with the existing code, or something I changed here. I do not think the other systems are relevant in this case.

pushfoo reviewed (changes requested):
If I'm not missing something, it looks like multiline doesn't work. Either something is wrong, or it's a limitation the doc needs to be clear about.

![image](https://github.com/pythonarcade/arcade/assets/36696816/3740c7b6-8b62-4c1f-bc83-c3e2eaef82a3)
```python
import arcade


class UsageAttempt(arcade.Window):

    def __init__(self, width: int = 320, height: int = 240):
        super().__init__(width=width, height=height)

        self.sprites = arcade.SpriteList()
        text_sprite = arcade.create_text_sprite(
            "First line\nsecond line",
            multiline=True,
            width=200,
        )
        text_sprite.position = self.width // 2, self.height // 2
        self.sprites.append(text_sprite)

    def on_draw(self):
        self.clear()
        self.sprites.draw()


if __name__ == "__main__":
    w = UsageAttempt()
    arcade.run()
```

Commit:
create_text_sprite

Commit:
//

Commit:
..

gran4 commented:
> If I'm not missing something, it looks like multiline doesn't work. Either something is wrong, or it's a limitation the doc needs to be clear about.

Working to fix it. Don't review yet. I don't know what the problem is.

Commit:
hi

bunny-therapist commented:
To me, the `fbo.clear((0, 0, 0, 255))` is causing the text to have a black box behind it. Should that really be there?

pushfoo commented:
> To me, the `fbo.clear((0, 0, 0, 255))` is causing the text to have a black box behind it. Should that really be there?

Please try swapping the 255 for a 0 if you have time. I need to look at some other PRs right now, and your help would be appreciated.

If replacing it works, the following steps would probably work improve the clarity of the code in addition to fixing the issue:
1. add `from arcade.color import TRANSPARENT_BLACK` to the imports
2. replace the tuple literal with `TRANSPARENT_BLACK` for clarity

bunny-therapist commented:
`fbo.clear((0,0,0,0))` works. I mean, it does not seem (to me) to make a difference adding it since the texture is already empty, so clearing it makes no difference, but it does not add a black box around the text.

Commit:
changes as sugested

Commit:
..

Commit:
small fixes

einarf commented on HI.py:
```diff
@@ -0,0 +1,84 @@
+    def __init__(self, width: int = 320, height: int = 240):
+        super().__init__(width=width, height=height)
+
+        self.sprites = arcade.SpriteList()
+        text_sprite = arcade.create_text_sprite(
+            "First line\nsecond line",
+            multiline=True,
+            width=200,
+        )
+        text_sprite.position = self.width // 2, self.height // 2
+        self.sprites.append(text_sprite)
+
+        self._label = pyglet.text.Label(
+            text="First line\nsecond line\nTHIRD NIGER",
+            x = 200,
+            y = 200,
+            font_name="Arial",
+            font_size=12,
+            anchor_x="left",
+            width=100,
+            align="baseline",
+            bold=True,
+            italic=True,
+            multiline=True,
+        )
+
+    def on_draw(self):
+        self.clear()
+        self.sprites.draw()
+
+        window = arcade.get_window()
+        with window.ctx.pyglet_rendering():
+            self._label.draw()
+
+
+
+
+if __name__ == "__main__":
+    w = UsageAttempt()
+    arcade.run()
```
This file should definitely not be in the PR.

einarf replied:
We only really allow test scripts in places like `arcade/experimental` and that's mainly things not in the root.

gran4 replied:
I already know. It is a test file to test the dev version. I will remove the file right before it is merged. Before that happens, the multiline must be fixed. I do not know the problem still.

gran4 committed:
.

einarf commented:
Thanks again for the PR. Since it contains `Hi.py` and liely need some more tweaks I will close this for now and see what I can salvage from it.
